cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -142,11 +142,10 @@ type ClusterConfiguration struct {
 // ControlPlaneComponent holds settings common to control plane component of the cluster
 type ControlPlaneComponent struct {
 	// ExtraArgs is an extra set of flags to pass to the control plane component.
-	// A key in this map is the flag name as it appears on the
-	// command line except without leading dash(es).
-	// TODO: This is temporary and ideally we would like to switch all components to
-	// use ComponentConfig + ConfigMaps.
-	ExtraArgs map[string]string
+	// An argument name in this list is the flag name as it appears on the
+	// command line except without leading dash(es). Extra arguments will override existing
+	// default arguments. Duplicate extra arguments are allowed.
+	ExtraArgs []Arg
```

Comment:
unclear if extra args will always remain as a config option.
removing the TODO to avoid confusion.